src/components/views/create/CreateProject.tsx

```diff
@@ -110,19 +109,26 @@ const CreateProject: FC<ICreateProjectProps> = ({ project }) => {
 		categories: storageCategories,
 		impactLocation: storageImpactLocation,
 		image: storageImage,
-		addresses: storageAddresses = {},
 	} = storageProjectData || {};
+	const storageAddresses =
+		storageProjectData?.addresses instanceof Array
+			? storageProjectData.addresses
+			: [];
 
 	const isDraft = project?.status.name === EProjectStatus.DRAFT;
 	const defaultImpactLocation = impactLocation || '';
-	const activeAddresses = addresses?.filter(a => a.isRecipient) || [];
-
-	const userAddresses = [...new Set(activeAddresses.map(a => a.address!))];
-
-	const addressesObj: { [key: number]: string } = {};
-	activeAddresses.forEach(a => {
-		addressesObj[a.networkId!] = a.address!;
-	});
+	const activeAddresses =
+		addresses
+			?.filter(a => a.isRecipient)
+			.map(a => {
+				const _a = { ...a };
+				delete _a.isRecipient;
+				return _a;
+			}) || [];
+
+	const userUniqueAddresses = [
+		...new Set(activeAddresses.map(a => a.address!)),
+	];
```

Comment:
Please revert changes are not part of solana support

Reply:
As I said before it's part of Solana. My decision is to use array instead of object for storing addresses, because we have this array type in backend as well and it's better not to change the type. Beside that I believe array fits better our case.
If I change addresses var type, I should change all other related components like WalletAddressInput and AddressInterface. Also I should change all address validations and test them all again, this way we may face lots of bugs. Although, I don't have enough time for that as I'm busy with other Solana issues.
1. if you think my code has bugs, please mention them and I will fix
2. if you have questions about any part of the code, we can have a quick call and I will explain them
3. if you think my code is against best practices, please tell me the reason and I will change it.
4. the code reviewer is responsible for reviewing the code functionality and test it thoroughly before merging. if you don't have time for this, please assign other devs as reviewer.

Reply:
cc @jainkrati

Reply:
@MohammadPCh to weigh in from frontend perspective here